extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -0,0 +1,222 @@
+    }
+  }
+
+  private static NumberDataPoint intDataPointToNumberDataPoint(IntDataPoint dataPoint)
+  {
+    return NumberDataPoint.newBuilder()
+            .setTimeUnixNano(dataPoint.getTimeUnixNano())
+            .setAsInt(dataPoint.getValue())
+            .addAllAttributes(dataPoint.getLabelsList()
+                    .stream()
+                    .map(label -> KeyValue.newBuilder()
+                            .setKey(label.getKey())
+                            .setValue(AnyValue.newBuilder().setStringValue(label.getValue()))
+                            .build())
+                    .collect(Collectors.toList()))
+            .build();
+  }
+
+  private InputRow parseNumberDataPoint(NumberDataPoint dataPoint,
+                                        String metricName,
+                                        List<KeyValue> resourceAttributes)
+  {
+    long timeUnixMilli = TimeUnit.NANOSECONDS.toMillis(dataPoint.getTimeUnixNano());
+    List<KeyValue> metricAttributes = labelsToAttributes(dataPoint);
+    Number value = getValue(dataPoint);
+
+    int capacity = resourceAttributes.size()
+            + metricAttributes.size()
+            + 2; // metric name + value columns
+
+    Map<String, Object> event = Maps.newHashMapWithExpectedSize(capacity);
+    event.put(this.metricDimension, metricName);
+    event.put(this.valueDimension, value);
+
+    resourceAttributes.stream()
+            .filter(ra -> this.attributeMembership.test(this.resourceAttributePrefix + ra.getKey()))
+            .forEach(ra -> event.put(this.resourceAttributePrefix + ra.getKey(), ra.getValue().getStringValue()));
+
+    metricAttributes.stream()
+            .filter(ma -> this.attributeMembership.test(this.metricAttributePrefix + ma.getKey()))
+            .forEach(ma -> event.put(this.metricAttributePrefix + ma.getKey(), ma.getValue().getStringValue()));
+
+    List<String> dimensions = this.schemaDimensions.isEmpty() ? new ArrayList<>(event.keySet()) : this.schemaDimensions;
+    return new MapBasedInputRow(timeUnixMilli, dimensions, event);
+  }
+
+  private static List<KeyValue> labelsToAttributes(NumberDataPoint dataPoint)
+  {
+    if (!dataPoint.getAttributesList().isEmpty()) {
+      return dataPoint.getAttributesList();
+    }
+    return dataPoint.getLabelsList()
+            .stream()
+            .map(label -> KeyValue.newBuilder()
+                    .setKey(label.getKey())
+                    .setValue(AnyValue.newBuilder().setStringValue(label.getValue()))
+                    .build())
+            .collect(Collectors.toList());
+  }
+
+  private static Number getValue(NumberDataPoint dataPoint)
+  {
+    if (dataPoint.hasAsInt()) {
+      return dataPoint.getAsInt();
+    } else if (dataPoint.hasAsDouble()) {
+      return dataPoint.getAsDouble();
+    } else {
+      throw new IllegalStateException("Unexpected dataPoint value type. Expected Int or Double");
+    }
```

Comment:
nit, it might be cleaner to write this as a switch statement using `dataPoint.getValueCase` to cover all the possible cases